lib/msal-node/src/config/Configuration.ts

```diff
@@ -67,6 +68,7 @@ export type NodeSystemOptions = {
     loggerOptions?: LoggerOptions;
     networkClient?: INetworkModule;
     proxyUrl?: string;
+    customAgentOptions?: CustomAgentOptions;
```

Comment:
1. Not sure about the prefix "custom". Maybe "httpAgentOptions" ?
2. Should we group proxyUrl and

Reply:
While I am in favor of better organization here, unfortunately `proxyUrl` is public at this point so we can't move it without a breaking change. If that's the direction we want to go we have two options:

1. Create a backlog item for Msal-Node v2 as a reminder to make this change when we're ready for v2
2. Create the new property bucket now in addition to the existing parameter, marking the existing parameter deprecated while maintaining the functionality until v2 when we can remove the old param.

I'm generally not in favor of double dipping on params though if we think there will be more options like this added in the near future it may be worth creating that new bucket now to get ahead of that

Reply:
I think we only expose `NetworkRequestOptions` for implementers of `INetworkModule`. But we do not expect ppl to set proxyUrl or agentOptions here. 

We expect customers to set proxyUrl via `SystemOptions`. And this is where we should expose httpsAgentOptions as well.

Architecturally, proxyUrl / httpAgentOptions should only be exposed on `SystemOptions` and injected in implementations of `INetworkModule` via the constuctor. They are not part of NetworkOptions. 

@tnorling @sameerag - what deprecation options do we have here? Can we add a comment that "proxyUrl" from `NetworkOptions` will not be used?

Reply:
`NetworkRequestOptions` is set by `msal-node`, not by developers. We only expose the type so that implementers of `INetworkModule` know what the object looks like when they need to use it but the contents of the object are controlled by us. All configuration values controlled by the developer go though the objects defined in this file.

Reply:
@bgavrilMS regarding #1, the agent options can be http or https, so I don't think it makes sense to call it httpAgentOptions.

Reply:
@bgavrilMS regarding #1, the agent options don't get converted into http vs https until the httpClient where we determine if the proxyUrl was passed in. Therefore, I don't think it makes sense to name it "httpAgentOptions" - it's only http when a proxy url has been specified.